### PR TITLE
add pytorch2onnx part

### PR DESCRIPTION
**Pull request by drcut**
[1] Add tools/pytorch2onnx.py, which can be used to convert Pytorch model into ONNX
[2] Modify the slide inference part to make it more clear and compatible with ONNX

### CODE REVIEW

File: tools/pytorch2onnx.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        input_all = [node.name for node in onnx_model.graph.input]
+        input_initializer = [
+            node.name for node in onnx_model.graph.initializer
+        ]
+        net_feed_input = list(set(input_all) - set(input_initializer))
+        assert (len(net_feed_input) == 1)
+        sess = rt.InferenceSession(output_file)
+        onnx_result = sess.run(
+            None, {net_feed_input[0]: img_list[0].detach().numpy()})[0]
+        if not (pytorch_result == onnx_result).all():
```
Comment (xvjiarui) [inline tools/pytorch2onnx.py:125]: We may use `np.allclose()` instead.

File: tools/pytorch2onnx.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Convert MMDet to ONNX')
+    parser.add_argument('config', help='test config file path')
+    parser.add_argument('--checkpoint', help='checkpoint file', default=None)
+    parser.add_argument('--show', action='store_true', help='show onnx graph')
+    parser.add_argument(
+        '--verify', action='store_true', help='verify the onnx model')
+    parser.add_argument('--output_file', type=str, default='tmp.onnx')
```
Comment (xvjiarui) [inline tools/pytorch2onnx.py:138]: `--output-file`

File: tools/pytorch2onnx.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    parser.add_argument('--opset_version', type=int, default=11)
```
Comment (xvjiarui) [inline tools/pytorch2onnx.py:139]: opset-version

File: tools/pytorch2onnx.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    segmentor = _convert_batchnorm(segmentor)
+
+    num_classes = segmentor.decode_head.num_classes
+
+    if args.checkpoint:
+        checkpoint = load_checkpoint(
+            segmentor, args.checkpoint, map_location='cpu')
+        # old versions did not save class info in checkpoints,
+        # this walkaround is for backward compatibility
+        if 'CLASSES' in checkpoint['meta']:
```
Comment (xvjiarui) [inline tools/pytorch2onnx.py:179]: For MMSeg, all checkpoints contain `CLASSES`.

File: mmseg/models/segmentors/encoder_decoder.py [hunk 5]
```diff
@@ -171,6 +172,8 @@ def slide_inference(self, img, img_meta, rescale):
         h_stride, w_stride = self.test_cfg.stride
         h_crop, w_crop = self.test_cfg.crop_size
         batch_size, _, h_img, w_img = img.size()
```
Comment (xvjiarui) [inline mmseg/models/segmentors/encoder_decoder.py:174]:
```suggestion
        batch_size, _, h_img, w_img = tuple(img.shape)
```
may do the trick.

File: tools/pytorch2onnx.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+        module_output.running_mean = module.running_mean
+        module_output.running_var = module.running_var
+        module_output.num_batches_tracked = module.num_batches_tracked
+    for name, child in module.named_children():
+        module_output.add_module(name, _convert_batchnorm(child))
+    del module
+    return module_output
+
+
+def _demo_mm_inputs(input_shape, num_classes):  # yapf: disable
```
Comment (hellock) [inline tools/pytorch2onnx.py:39]: Is `# yapf: disable` necessary?

**Comment by hellock** [comment]
LGTM.

**Pull request merged**